python/ray/dataframe/io.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from itertools import chain
+from io import BytesIO
+import os
+
+from pyarrow.parquet import ParquetFile
+import pandas as pd
+
+from .dataframe import from_pandas, ray, DataFrame
+from . import get_nprtitions
+
+
+def read_parquet(path, engine='auto', columns=None, **kwargs):
+    """Load a parquet object from the file path, returning a DataFrame.
+    Ray DataFrame only supports pyarrow engine for now.
+
+    Args:
+        path: The filepath of the parquet file.
+              We only support local files for now.
+        engine: Ray only support pyarrow reader. This argument doesn't do anything for now. 
+        kwargs: Pass into parquet's read_row_group function.
+    """
+    pf = ParquetFile(path)
+    n_row_groups = pf.metadata.num_row_groups
+    partition_per_row_group = get_nprtitions() // n_row_groups
+    ray_row_groups = [
+        _read_parquet_row_group.remote(path, columns, i,
+                                       partition_per_row_group, kwargs)
+        for i in range(n_row_groups)
+    ]
+    return _vertical_concat(ray.get(ray_row_groups))
```

Comment:
I think we wouldn't need this here if the remote functions just returned a `pd.DataFrame` and you built the `ray.DataFrame` from the list of those. It should be a bit more efficient, plus there are some convenient things you can do now to optimize some performance. For example, now you can set `_lengths` (after DataFrame creation) because you already know it here (`df._lengths = [row_group_size for _ in n_row_groups]`).

Reply:
We do need to join ray_df here because this function runs locally. We don't want ray to transfer all df from object store to local machine memory. This is especially true for larger than memory dataframe. If we construct ray df here, local machine must be able to handle large dataframe; the current approach let each remote worker handles a subset of the dataframe.